test/swivel/Swivel.sol

```diff
@@ -821,6 +823,9 @@ contract Swivel is ISwivel {
       // Euler withdraw is void
       IEuler(c).withdraw(0, a);
       return true;
+    } else if (p == uint8(Protocols.Lido)) {
+      ILido wstEth = ILido(c);
+      return wstEth.unwrap(a / wstEth.getStETHByWstETH(1e18)) >= 0;
```

Comment:
I'm not sure if this is the right call - we may be able to get away with just passing the entirety of the balance or `a` (assuming `a` represents the user's amount of tokens they would like to withdraw.
```suggestion
      return wstEth.unwrap(IErc20(address(c).balanceOf(msg.sender or a)) >= 0;
```

Reply:
I could be wrong but it looks like [`unwrap`](https://docs.lido.fi/contracts/wsteth#unwrap) takes in the amount of tokens weighted staked ETH amount.

Reply:
@JTraversa bring some prod knowledge in here... 
this pass i had assumed it was like the current yearn solution.

Reply:
Unwrap seems to accept the wstETH balance as its param, in other words `shares` of the wstETH vault (compounding token amounts), rather than raw `stETH` amounts (underlying token amounts) 

So using the underlying amount wouldnt work (which is why i think is being suggested?)
_not 100% what you mean by "weighted staked eth amount"_

Reply:
I don't understand why we don't just call `unwrap` on the amount of tokens (`a`). If `a` is the number of wstETH tokens, then this would be simpler than calculating the rate.

Reply:
@sourabhmarathe 
Your first comment:
> (assuming a represents the user's amount of tokens they would like to withdraw)

New comment: 

> If a is the number of wstETH tokens

So thats where your confusion is.

`a` is the user's amount of tokens they would like to withdraw, not the amount of `shares` (wstETH)